app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php

```diff
@@ -0,0 +1,193 @@
+      'required' => true,
+      'allowEmpty' => false
+    ),
+    'co_person_id' => array(
+      'rule' => 'numeric',
+      'required' => true,
+      'allowEmpty' => false
+    ),
+    'type' => array(
+      'content' => array(
+        'rule' => array('validateExtendedType',
+          array('attribute' => 'EmailAddress.type',
+                'default' => array(EmailAddressEnum::Delivery,
+                                   EmailAddressEnum::Forwarding,
+                                   EmailAddressEnum::MailingList,
+                                   EmailAddressEnum::Official,
+                                   EmailAddressEnum::Personal,
+                                   EmailAddressEnum::Preferred,
+                                   EmailAddressEnum::Recovery))),
+        'required' => true,
+        'allowEmpty' => false
+      )
+    ),
+  );
+
+  /**
+   * Create the email and history recordw
+   *
+   * @since  COmanage Registry v4.1.0
+   * @param  string $token     Token used for verification
+   * @return integer           ID of the new EmailAddress Record
+   * @throws RuntimeException
+   */
+
+  public function addEmailToPerson($token, $actorCoPersonId) {
+    // Retrieve the Verification record and the configuration
+    $rec = $this->getRecordToVerify($token);
+
+    // Create the new CO Person Email record
+    $emailAttrs = array(
+      'mail' => $rec['EmailAddressWidgetVerification']['email'],
+      'type' => $rec['EmailAddressWidgetVerification']['type'],
+      'verified' => true,
+      'co_person_id' => $rec['EmailAddressWidgetVerification']['co_person_id']
+    );
+
+    try {
+      $EmailAddress = ClassRegistry::init('EmailAddress');
+      if(!$EmailAddress->save($emailAttrs, array("provision" => true,
+                                                 "trustVerified" => true))) {
+        throw new RuntimeException(_txt('er.db.save'));
+      }
+
+      $CoPerson = ClassRegistry::init('CoPerson');
+      // History record for the new record
+      $CoPerson->HistoryRecord->record($rec['EmailAddressWidgetVerification']['co_person_id'],
+                                       null,
+                                       null,
+                                       $actorCoPersonId,
+                                       ActionEnum::CoPersonEditedManual,
+                                       _txt('pl.emailaddresswidget.rs.added-a', array($rec['EmailAddressWidgetVerification']['email'])));
+
+      // History Record for the verification
+      $CoPerson->HistoryRecord->record($rec['EmailAddressWidgetVerification']['co_person_id'],
+                                       null,
+                                       null,
+                                       $actorCoPersonId,
+                                       ActionEnum::EmailAddressVerified,
+                                       _txt('pl.emailaddresswidget.rs.mail.verified', array($rec['EmailAddressWidgetVerification']['email'])));
```

Comment:
I think we could collapse these into a single message: "Added Verified Email Address %1$s"